devito/operator.py

```diff
@@ -100,6 +101,9 @@ def __init__(self, nt, shape, dtype=np.float32, stencils=[],
         for d in dimensions:
             sym_undef.remove(d)
 
+        # Replace all occurences of the `time` dimension with `t`
+        self.stencils = [eqn.subs(time, t) for eqn in self.stencils]
+
```

Comment:
Good catch, I'm not sure. It doesn't break any existing tests, but I'll have closer look.